x-pack/test/api_integration/apis/transform/transforms_update.ts

```diff
@@ -0,0 +1,150 @@
+    const config = generateTransformConfig(transformId);
+    await transform.api.createTransform(transformId, config);
+  }
+
+  function getTransformUpdateConfig() {
+    return {
+      source: {
+        index: 'ft_*',
+        query: {
+          term: {
+            airline: {
+              value: 'AAL',
+            },
+          },
+        },
+      },
+      description: 'the-updated-description',
+      dest: {
+        index: 'user-the-updated-destination-index',
+      },
+      frequency: '60m',
+    };
+  }
+
+  describe('/api/transform/transforms/{transformId}/_update', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await transform.testResources.setKibanaTimeZoneToUTC();
+      await createTransform('the-transform-1');
+    });
+
+    after(async () => {
+      await transform.api.cleanTransformIndices();
+    });
+
+    it('should update a transform', async () => {
+      // assert the original transform for comparison
+      const { body: transformOriginalBody } = await supertest
+        .get('/api/transform/transforms/the-transform-1')
+        .auth(
+          USER.TRANSFORM_POWERUSER,
+          transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+        )
+        .set(COMMON_REQUEST_HEADERS)
+        .send()
+        .expect(200);
+
+      expect(transformOriginalBody.count).to.eql(expected.transformOriginalConfig.count);
+      expect(transformOriginalBody.transforms).to.have.length(
+        expected.transformOriginalConfig.count
+      );
+
+      const transformOriginalConfig = transformOriginalBody.transforms[0];
+      expect(transformOriginalConfig.id).to.eql(expected.transformOriginalConfig.id);
+      expect(transformOriginalConfig.source).to.eql(expected.transformOriginalConfig.source);
+      expect(transformOriginalConfig.description).to.eql(undefined);
+      expect(transformOriginalConfig.settings).to.eql({});
+
+      // update the transform and assert the response
+      const { body: transformUpdateResponseBody } = await supertest
+        .post('/api/transform/transforms/the-transform-1/_update')
+        .auth(
+          USER.TRANSFORM_POWERUSER,
+          transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+        )
+        .set(COMMON_REQUEST_HEADERS)
+        .send(getTransformUpdateConfig())
+        .expect(200);
+
+      const expectedUpdateConfig = getTransformUpdateConfig();
+      expect(transformUpdateResponseBody.id).to.eql(expected.transformOriginalConfig.id);
+      expect(transformUpdateResponseBody.source).to.eql({
+        ...expectedUpdateConfig.source,
+        index: ['ft_*'],
+      });
+      expect(transformUpdateResponseBody.description).to.eql(expectedUpdateConfig.description);
+      expect(transformUpdateResponseBody.settings).to.eql({});
+
+      // assert the updated transform for comparison
+      const { body: transformUpdatedBody } = await supertest
```

Comment:
We're validating the response body for the update request here, which is good. But I'm wondering if we would also (or instead?) want to validate the actual transform after update (via an additional `get transform`), just to make sure that the changes have been persisted correctly.

Reply:
That's what the code is already doing :) In the previous section after `// update the transform and assert the response` we assert the response body of the update call, the section you added the comment to after `// assert the updated transform for comparison` gets the updated config again and tests if the update was persisted.

Reply:
Yes, correct, seems I've mixed up the requests, sorry for the noise.